src/components/common/SearchBar.tsx

```diff
@@ -1,60 +1,158 @@
+  };
+
+  const handleDropDownItem = (item: string) => {
+    setInputValue(item);
+    setIsInputValue(false);
+    setItemIndex(-1);
+    handleSearch(item);
+  };
+
+  const handleSearch = (searchValue: string) => {
+    if (searchValue.trim()) {
+      const query = encodeURIComponent(searchValue);
+      navigate(`/search?query=${query}`);
+    }
+  };
+
+  const handleDropDownKey = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (isInputValue) {
+      if (event.key === 'ArrowDown' && dropDownList.length - 1 > itemIndex) {
+        setItemIndex(itemIndex + 1);
+      }
+
+      if (event.key === 'ArrowUp' && itemIndex >= 0) {
+        setItemIndex(itemIndex - 1);
+      }
+
+      if (event.key === 'Enter') {
+        event.preventDefault();
+        if (itemIndex >= 0) {
+          handleDropDownItem(dropDownList[itemIndex]);
+        } else {
+          handleSearch(inputValue);
+        }
+      }
     }
   };
 
+  useEffect(showDropDownList, [inputValue]);
+
   return (
     <SearchBarContainer>
-      <SearchInputWrapper>
-        <SearchInput type="text" value={inputValue} onChange={handleInputChange} placeholder={placeholder} />
+      <SearchInputWrapper $isInputValue={isInputValue}>
+        <SearchInput
+          type="text"
+          value={inputValue}
+          onChange={handleInputChange}
+          onKeyDown={handleDropDownKey}
+          placeholder={placeholder}
+        />
+        <SearchIconWrapper onClick={() => handleSearch(inputValue)} />
       </SearchInputWrapper>
-      <SearchIconWrapper />
+      {isInputValue && (
+        <SearchDropDownBox>
+          {dropDownList.length === 0 ? (
+            <SearchDropDownItem>해당하는 키워드가 없습니다!</SearchDropDownItem>
+          ) : (
+            dropDownList.map((item, index) => {
+              const matchIndex = item.toLowerCase().indexOf(normalizedInput.toLowerCase());
+              return (
+                <SearchDropDownItem
+                  key={item}
+                  onClick={() => handleDropDownItem(item)}
+                  onMouseOver={() => setItemIndex(index)}
+                  className={itemIndex === index ? 'selected' : ''}
+                >
+                  {matchIndex !== -1 ? (
+                    <>
+                      {item.substring(0, matchIndex)}
+                      <span style={{ color: 'red' }}>
+                        {item.substring(matchIndex, matchIndex + normalizedInput.length)}
+                      </span>
+                      {item.substring(matchIndex + normalizedInput.length)}
+                    </>
+                  ) : (
+                    item
+                  )}
+                </SearchDropDownItem>
+              );
+            })
+          )}
+        </SearchDropDownBox>
+      )}
     </SearchBarContainer>
   );
 }
 
 const SearchBarContainer = styled.div`
+  width: 100%;
+  height: 44px;
+`;
+
+const SearchInputWrapper = styled.div<{ $isInputValue: boolean }>`
```

Comment:
```suggestion
const SearchInputWrapper = styled.div<{ isInputValue: boolean }>`
```

다른 컴포넌트와 일관되도록 `$` prefix 없이 컴포넌트 prop 이름을 네이밍하는게 좋을 거 같습니다.

Reply:
렌더링 개선과 함께 일괄 수정 했습니다!